pkg/arvo/sys/vane/gall.hoon

```diff
@@ -958,8 +958,18 @@
         moves        moves
       ==
     ::
+    ++  ap-yawn-all
+      ^-  (list card:agent)
+      %-  zing
+      %+  turn  ~(tap by ken.yoke)
+      |=  [=spar:ames wyz=(set wire)]
+      %+  turn  ~(tap in wyz)
+      |=  =wire
+      [%pass wire %arvo %a %yawn spar]
+    ::
     ++  ap-idle
       ?:  ?=(%| -.agent.yoke)  ap-core
+      =>  (ap-ingest ~ |.([ap-yawn-all p.agent.yoke]))
```

Comment:
If an agent gets `%idle`d then woken back up again, does it get notified?  I don't think it does.  Without knowing that it got turned off and back on, if we kill its outbound `keen`s when it gets idled, the agent won't know they got killed, so it might sit there forever without knowing it needs to re-send those requests.

Reply:
I'm discussing this with @joemfb right now.  I think the best way to handle idling is for Gall to cancel the outgoing scry requests on idle, hold onto the request wires and spars in its state, and then when the agent is turned back on, re-send those requests to Ames.  This should work fine as long as Gall doesn't crash if it happens to receive a scry response in between idling the agent and the agent turning back on -- although Ames itself should handle that, since it remembers which outgoing scry requests are live.

That way the agent doesn't need to be notified, and we don't artificially kill perfectly legitimate requests, but we also avoid sending superfluous network traffic to repeat requests forever on behalf of an idled agent.

Reply:
I actually thought about this too, but figured that since we were planning to include keens in the bowl it was fine. The agent could discover it that way. But I agree that this is more polite. (And of course, we're not doing keens in the bowl at this time)